### DISCUSSION TITLE
How can I save pages getting the same results of extension

### DISCUSSION THREAD

**Discussion by dav-dav-82** [gildas-lormeau/single-file-cli#140] [Q&A] [answered]
I would like to batch save a list of pages from a list of URLs.
So I tried to use the CLI tool, but, the result is pages full of cookie requests, even some servers refuse to load for "network policies" and so on.
The target result I want is the one I get by using Firefox and my user profile.
How do I get there?
Thank you in advance!

**Comment by gildas-lormeau** [comment]
`single-file` uses a Chromium browser (today) to save web pages so it won't be able to import the user data as-is, unfortunately. FYI, if the user data was stored in a Chromium profile folder, you could use `--browser-arg --user-data-dir /path/to/the/user-data`. Alternatively, you could pass cookies via the option `--browser-cookies-file` and exporting cookies from firefox into a text file.

**Reply by dav-dav-82** [reply]
Thank you very much for your kind answer!
Understood.
I just saw on the "batch page" of the extension that you are providing a button where I can put URLs on each line!
I think this is super useful for my use case!
Maybe the only problem I see is that all the URLs given are opened at once on different tabs. So, if I have hundreds of URLs it gets hard to manage from a resource point of view and sometimes you get into "too many requests" problem.
If the URLs were managed in chunks (let's say in group of 5-10) and a slight delay could be introduced, that would be perfect!
Would you maybe consider this feature?

**Reply by gildas-lormeau** [reply, marked as answer]
That's strange, the URLs should be managed by chunks in the extension, as you described. You can define the chunk size by exporting the options and setting the `maxParallelWorkers` value, see https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options for more info.
